src/components/HomePageContent/HomePageContainer.tsx

```diff
@@ -36,12 +35,14 @@ export const HomePageContainer = (): React.ReactElement => {
     }
   });
   React.useEffect(() => {
-    if (location.search !== "") {
-      const queryParams = queryString.parse(location.search);
-      const anchorStr = decodeURIComponent(location.hash.substr(1));
+    const { query, asPath } = router;
+    if (Object.values(query).length > 0) {
+      const hash = asPath.split("#")[1];
+
+      const anchorStr = decodeURIComponent(hash);
       const anchor = anchorStr ? JSON.parse(anchorStr) : {};
       dispatch(resetCertificateState());
-      const action = JSON.parse(queryParams.q as string);
+      const action = JSON.parse(query.q as string);
```

Comment:
what's the type of query.q ? (why need casting)

Reply:
query.q is string | string[] | undefined. type error will be thrown if its not casted